setup.py

```diff
@@ -165,466 +26,206 @@ def _get_interop_filename():
     return os.path.join("src", "runtime", interop_filename)
 
 
-def _get_source_files():
-    """Walk project and collect the files needed for ext_module"""
-    for ext in (".sln",):
-        for path in glob.glob("*" + ext):
-            yield path
+def _write_configure_props():
+    defines = [
+        "PYTHON{0}{1}".format(PY_MAJOR, PY_MINOR),
+    ]
 
-    for root, dirnames, filenames in os.walk("src"):
-        for ext in (".cs", ".csproj", ".snk", ".config", ".py", ".c", ".h", ".ico"):
-            for filename in fnmatch.filter(filenames, "*" + ext):
-                yield os.path.join(root, filename)
+    if sys.platform == "win32":
+        defines.append("WINDOWS")
 
-    for root, dirnames, filenames in os.walk("tools"):
-        for ext in (".exe", ".py", ".c", ".h"):
-            for filename in fnmatch.filter(filenames, "*" + ext):
-                yield os.path.join(root, filename)
+    if hasattr(sys, "abiflags"):
+        if "d" in sys.abiflags:
+            defines.append("PYTHON_WITH_PYDEBUG")
+        if "m" in sys.abiflags:
+            defines.append("PYTHON_WITH_PYMALLOC")
 
+    # check the interop file exists, and create it if it doesn't
+    interop_file = _get_interop_filename()
+    if not os.path.exists(interop_file):
+        print("Creating {0}".format(interop_file))
+        geninterop = os.path.join("tools", "geninterop", "geninterop.py")
+        check_call([sys.executable, geninterop, interop_file])
 
-def _get_long_description():
-    """Helper to populate long_description for pypi releases"""
-    return open("README.rst").read()
+    import xml.etree.ElementTree as ET
 
+    proj = ET.Element("Project")
+    props = ET.SubElement(proj, "PropertyGroup")
+    f = ET.SubElement(props, "PythonInteropFile")
+    f.text = os.path.basename(interop_file)
 
-def _update_xlat_devtools():
-    global DEVTOOLS
-    if DEVTOOLS == "MsDev":
-        DEVTOOLS = "MsDev15"
-    elif DEVTOOLS == "Mono":
-        DEVTOOLS = "dotnet"
+    c = ET.SubElement(props, "ConfiguredConstants")
+    c.text = " ".join(defines)
 
+    ET.ElementTree(proj).write(CONFIGURED_PROPS)
```

Comment:
Is it actually OK to create a new file in the current directory when installing a Python package?

Reply:
I think so. Modern Python packaging (PEP517/518) runs in isolation in a temporary directory, if I understood things correctly.

Reply:
If that would be so, I would not expect the need for `configured.props` in `.gitignore`. Something seems amiss.

Reply:
That is for development.